core/src/datayoga_core/blocks/relational/write/block.py

```diff
@@ -91,15 +84,21 @@ def init(self, context: Optional[Context] = None):
         self.tbl = sa.Table(self.table, sa.MetaData(schema=self.schema), autoload_with=self.engine)
 
         if self.opcode_field:
-            business_keys = get_column_mapping(self.keys)
+            business_key_columns = [column["column"] for column in get_column_mapping(self.keys)]
+            mapping_columns = [column["column"] for column in get_column_mapping(self.mapping)]
+
+            columns = business_key_columns + [x for x in mapping_columns if x not in business_key_columns]
+
+            for column in columns:
+                if not column in self.tbl.columns:
+                    raise ValueError(f"{column} column does not exist in {self.tbl.fullname} table")
 
-            self.delete_stmt = self.tbl.delete(
-                sa.and_(*[sa.text(f"{self.tbl.columns.get(field['column'])} = {sa.bindparam(field['key'])}")
-                          for field in business_keys]))
+            self.delete_stmt = self.tbl.delete().where(
+                sa.and_(*[(self.tbl.columns[column] == sa.bindparam(column)) for column in business_key_columns]))
```

Comment:
there is an edge case where you have a composite key and one of the fields is null.
it then requires:
key=incoming.key or key is null and incoming.key is null. I can open an issue for this. not urgent since we are relying on "on conflict" which requires not null constraints anyways.